applications/dashboard/settings/structure.php

```diff
@@ -765,6 +766,69 @@
     ->column('UpdateIPAddress', 'ipaddress', true)
     ->set($Explicit, $Drop);
 
+// If the AllIPAddresses column exists, attempt to migrate legacy IP data to the UserIP table.
+if ($AllIPAddressesExists) {
+    $limit = 10000;
+    $resetBatch = 100;
+
+    // Grab the initial batch of users.
+    $legacyIPAddresses = $SQL->select(['UserID', 'AllIPAddresses', 'InsertIPAddress', 'LastIPAddress', 'DateLastActive'])
+        ->from('User')->where('AllIPAddresses is not null')->limit($limit)
+        ->get()->resultArray();
+
+    do {
+        $processedUsers = [];
+
+        // Iterate through the records of users with data needing to be migrated.
+        foreach ($legacyIPAddresses as $currentLegacy) {
+            // Pull out and format the relevant bits, where necessary.
+            $allIPAddresses = explode(',', $currentLegacy['AllIPAddresses']);
+            $dateLastActive = val('DateLastActive', $currentLegacy);
+            $insertIPAddress = val('InsertIPAddress', $currentLegacy);
+            $lastIPAddress = val('LastIPAddress', $currentLegacy);
+            $userID = val('UserID', $currentLegacy);
+
+            // If we have a LastIPAddress record, use it.  Give it a DateUpdated of the user's DateLastActive.
+            if (!empty($lastIPAddress)) {
+                Gdn::userModel()->saveIP(
+                    $userID,
+                    $lastIPAddress,
+                    $dateLastActive
+                );
+            }
+
+            // Only save InsertIPAddress if it differs from LastIPAddress and is in AllIPAddresses (to avoid admin IPs).
+            if ($insertIPAddress !== $lastIPAddress && in_array($insertIPAddress, $allIPAddresses)) {
+                Gdn::userModel()->saveIP(
+                    $userID,
+                    $insertIPAddress
+                );
+            }
+
+            // Record the processed user's ID.
+            $processedUsers[] = $userID;
+
+            // Every X records (determined by $resetBatch), clear out the AllIPAddresses field for processed users.
+            if (count($processedUsers) > 0 && (count($processedUsers) % $resetBatch) === 0) {
+                $this->SQL->update('User')->set('AllIPAddresses', null)->whereIn('UserID', $processedUsers)
```

Comment:
Usually if a SQL chain needs multiple lines, it's preferable to just break it at every `->` and stack them.